### PR TITLE
chore: onboard onto unified pipeline

### PR DESCRIPTION
**Pull request by rzhao271**
Ref https://github.com/microsoft/vscode-internalbacklog/issues/3191

### CODE REVIEW

**Comment by rzhao271** [comment]
I noticed CI wasn't running on here, so I added a GitHub Action. Let me know if I should be using Azure DevOps for CI as well.

**Review by deepak1556 — CHANGES_REQUESTED** [review]
Yes, please use Azure pipelines for the PR runs as well.

File: pipelines/publish.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+          # the rest of the build steps are part of the 'prepack' script, automatically run when the pipeline invokes 'yarn pack'
+
+        testPlatforms:
+          - name: Linux
+            nodeVersions:
+              - 16.x
+          - name: MacOS
+            nodeVersions:
+              - 16.x
```
Comment (deepak1556) [inline pipelines/publish.yml:41]: Linux and MacOS are not needed for this module

File: binding.gyp [hunk 2]
```diff
@@ -13,12 +13,18 @@
...
+            '/w34267',
+            '/sdl'
           ]
         },
         'VCLinkerTool': {
           'AdditionalOptions': [
-            '/guard:cf'
+            '/guard:cf',
+            '/CEFCOMPAT',
+            '/DYNAMICBASE'
```
Comment (deepak1556) [inline binding.gyp:27]: Can you adjust the flags here similar to https://github.com/microsoft/vscode-windows-ca-certs/pull/7#pullrequestreview-1296666210

**Review by deepak1556 — APPROVED** [review]

**Comment by deepak1556** [comment]
Oh looks like the pipelines haven't been triggered.

**Comment by deepak1556** [comment]
You can trigger a run from the branch based on the new pipeline path, triggered at https://dev.azure.com/monacotools/Monaco/_build/results?buildId=205459&view=results

**Review by deepak1556 — APPROVED** [review]

**Pull request merged**